src/main/java/org/gradle/profiler/BuildToolCommandLineScenarioDefinition.java

```diff
@@ -33,9 +34,14 @@ protected void printDetail(PrintStream out) {
         out.println("  Targets: " + getTargets());
     }
 
-    public String getExecutablePath() {
+    public String getExecutablePath(File projectDir) {
         String toolHomePath = getToolHome() == null ? System.getenv(getToolHomeEnvName()) : getToolHome().getAbsolutePath();
-        return toolHomePath == null ? getExecutableName() : toolHomePath + "/bin/" + getExecutableName();
+        return toolHomePath == null ? getExecutablePathWithoutToolHome(projectDir) : toolHomePath + "/bin/" + getExecutableName();
+    }
+
+    @OverridingMethodsMustInvokeSuper
+    protected String getExecutablePathWithoutToolHome(File projectDir) {
```

Comment:
🤔 What if we pass projectDir to `getExecutableName()` (and maybe we rename it?) so we have just one method?

Reply:
We also use `getExecutableName()` when looking up the executable from the tool home.